### ISSUE TITLE
[Question] Optimization of "reshape_as" operation

### ISSUE THREAD

**Issue by dhbloo** [inisis/OnnxSlim#95]
Hi, thanks for your great work!

I have an RL model which outputs a 2D-shaped policy that needs to be normalized. After I flatten the policy to 1D and apply Softmax on it, reshaping it back causes the graph to be very messy with many unnecessary slice and concat nodes. 

Original pytorch code:
```
# policy.shape=(batch_size, 1, height, width)
policy = policy.squeeze(1)
policy_flat = torch.flatten(policy, start_dim=1)
policy_flat = torch.nn.functional.softmax(policy_flat, dim=1)
policy = torch.reshape(policy_flat, policy.shape)
```

The original graph:
<img width="335" alt="Image" src="https://github.com/user-attachments/assets/89b172d2-6071-4c60-888e-a50804c67726" />

Graph after using onnxslim:
<img width="339" alt="Image" src="https://github.com/user-attachments/assets/32919132-2d6d-4dd2-97d0-1acd60976b35" />

I wonder if it is possible that a new pass can be added to optimize this pattern? I have looked at the code a little bit but I am not sure what's is the best way to achieve this. I wonder if you could provide some suggestions.

**Comment by inisis** [comment]
Hi, thanks for your interest in our project, it seems the shape inference failed, and your captured screen seems to be part of the model, normally the shape inference should work, is it possible to provide me your onnx model, you can send it to desmond.yao@buaa.edu.cn.

**Comment by dhbloo** [comment]
The model does have a dynamic-shaped input, so maybe the constant folding logic can not be used here. But I still think it should be possible to eliminate the "gather-unsqueeze-concat" operation here, but I am not sure how to achieve that. (I do not understand why PyTorch split such graph output in the first place).

I have sent you the ONNX model file, you can have a look at it. Thanks!

**Comment by inisis** [comment]
I agree with you, and will add one optimization pass soon

**Comment by inisis** [comment]
You can try pip install git+https://github.com/inisis/OnnxSlim@dev, this should achieve what you want

**Comment by dhbloo** [comment]
I have tested the new version, and it perfectly optimizes that part.
Thanks a lot!
